src/components/form/index.tsx

```diff
@@ -1,15 +1,51 @@
-import React from 'react';
+import React, { forwardRef, useImperativeHandle, useState } from 'react';
 import { Container, Input, LabelInput } from './style';
+import { ErrorMessage } from './style';
+
 interface FormProps {
   name: string;
-  info: string;
-  setValue?: (text: string) => void;
+  validateValue: (text: string) => string | null;
```

Comment:
normalmente em props do tipo função costumamos utilizar o prefixo `on` (acho que é bem comum isso no react no geral 🤔), isso facilita identificar o que é uma função e o que é apenas um valor, nesse caso aqui o ideal seria 
```suggestion
  onValidateValue: (text: string) => string | null;
```